app/controllers/course/controller.rb

```diff
@@ -9,16 +9,23 @@ class Course::Controller < ApplicationController
   #
   #   {
   #      title: 'Sidebar Item Title'
+  #      type: :admin # Will be considered as `:normal` if not set
+  #      path: path_to_the_component
   #      unread: 0 # or nil
   #   }
   #
   # The elements are rendered on all Course controller subclasses as part of a nested template.
-  def sidebar
+  # @param [Symbol] type The type of sidebar item, all sidebar items will be returned if the type
+  # is not specified
+  # @return [Array] The array of sidebar items of the given type
+  def sidebar(type = nil)
     array_of_component_arrays = current_component_host.components.map do |component|
       component.get_sidebar_items(self)
     end
 
-    array_of_component_arrays.tap(&:flatten!)
+    sidebar_items = array_of_component_arrays.tap(&:flatten!)
```

Comment:
`select!` mutates the array in place, so no need to double memory usage. `select` makes a copy. The latter is more functional.